src/components/toggle/toggle.controller.ts

```diff
@@ -57,35 +69,36 @@ export class IgcToggleController implements ReactiveController {
     this.update();
   }
 
-  public hostConnected() {
-    this.addEventListeners();
-  }
-
   public hostDisconnected() {
-    this.removeEventListeners();
+    this.abortController.abort();
   }
 
   public update() {
     this.toggleDirective = igcToggle(this.target, this.host, this);
-    this.addEventListeners();
+    this.addListeners();
   }
 
   protected hide() {
     this._hide ? this._hide() : this.host.hide();
   }
 
-  private addEventListeners() {
-    if (this.host.open) {
-      document.addEventListener('scroll', this.handleScroll, true);
-      if (!this.host.keepOpenOnOutsideClick) {
-        document.addEventListener('click', this.documentClicked, true);
-      }
+  private setupDocumentListeners() {
+    const options: AddEventListenerOptions = {
+      capture: true,
+      signal: this.abortController.signal,
+    };
+
+    if (!this.host.keepOpenOnOutsideClick) {
+      document.addEventListener('click', this.documentClicked, options);
     }
+
+    document.addEventListener('scroll', this.handleScroll, options);
   }
 
-  private removeEventListeners() {
-    document.removeEventListener('click', this.documentClicked, true);
-    document.removeEventListener('scroll', this.handleScroll, true);
+  private addListeners() {
```

Comment:
Minor nit: Kind of weird this being called `addListeners` when it both adds and removes them now.